docs/data-workarounds.md

```diff
@@ -0,0 +1,180 @@
+## Guide: Adding a Data Workaround 
+
+When generating `api-definitions`, unexpected behavior can occur that requires a data workaround to be added. This can be because a type is not what we expect, attributes like `Computed` need to be added, or whole attributes need to be hand defined.
+
+These data workarounds live in `/tools/importer-rest-api-specs/internal/components/apidefinitions/parser/dataworkarounds` and this guide will walk through how to add a new one.
+
+## Open an issue on github.com/Azure/azure-rest-api-specs/issues 
+
+If there is an issue with the api-specs, open an issue on that repo as we'll use the issue number in our filename.
+
+## Create a new workaround in `/tools/importer-rest-api-specs/internal/components/apidefinitions/parser/dataworkarounds`
+
+The file should be called `workaround_servicepackage_issue#`. For this example, we'll create a file called `workaround_web_31682.go`
+
+The following is boilerplate that uses the package and api issue number that this workaround applies to
+```
+var _ workaround = workaroundWeb31682{}
+
+type workaroundWeb31682 struct {
+}
+
+// This checks that this workaround will only apply to a specific service/api version
+func (workaroundWeb31682) IsApplicable(serviceName string, apiVersion sdkModels.APIVersion) bool {
+	return serviceName == "Web" && apiVersion.APIVersion == "2016-06-01"
+}
+
+func (workaroundWeb31682) Name() string {
+	return "Web / 31682"
+}
+```
+
+Now we can get into the meat and potatoes of how to override this specific issue. 
+
+I'm using [31682](https://github.com/Azure/azure-rest-api-specs/issues/31682) in this example where we're unmarshalling Parameter Value incorrectly and we're seeing `Error: occurred unmarshalling JSON - Error = 'json: cannot unmarshal bool into Go struct field` in the provider.
```

Comment:
I would provide this information at the start of the instruction to give context on the example that we're using for the guide, see my comment further up
```suggestion
```